plugins/importexport/native/NativeImportExportPlugin.inc.php

```diff
@@ -228,7 +229,144 @@ function importSubmissions($importXml, $deployment) {
+					$this->usage($scriptName);
+					return;
+				}
+
+				$filter = 'native-xml=>issue';
+				// is this articles import:
+				$xmlString = file_get_contents($xmlFile);
+				$document = new DOMDocument();
+				$document->loadXml($xmlString);
+				if (in_array($document->documentElement->tagName, array('article', 'articles'))) {
+					$filter = 'native-xml=>article';
+				}
+				$deployment = new NativeImportExportDeployment($press, $user);
+				$deployment->setImportPath(dirname($xmlFile));
+				$content = $this->importSubmissions($xmlString, $filter, $deployment);
+				$validationErrors = array_filter(libxml_get_errors(), function($a) {
+					return $a->level == LIBXML_ERR_ERROR || $a->level == LIBXML_ERR_FATAL;
+				});
+
+				// Are there any import warnings? Display them.
+				$errorTypes = array(
+					ASSOC_TYPE_ISSUE => 'issue.issue',
+					ASSOC_TYPE_SUBMISSION => 'submission.submission',
+					ASSOC_TYPE_SECTION => 'section.section',
+				);
+				foreach ($errorTypes as $assocType => $localeKey) {
+					$foundWarnings = $deployment->getProcessedObjectsWarnings($assocType);
+					if (!empty($foundWarnings)) {
+						echo __('plugins.importexport.common.warningsEncountered') . "\n";
+						$i = 0;
+						foreach ($foundWarnings as $foundWarningMessages) {
+							if (count($foundWarningMessages) > 0) {
+								echo ++$i . '.' . __($localeKey) . "\n";
+								foreach ($foundWarningMessages as $foundWarningMessage) {
+									echo '- ' . $foundWarningMessage . "\n";
+								}
+							}
+						}
+					}
+				}
+
+				// Are there any import errors? Display them.
+				$foundErrors = false;
+				foreach ($errorTypes as $assocType => $localeKey) {
+					$currentErrors = $deployment->getProcessedObjectsErrors($assocType);
+					if (!empty($currentErrors)) {
+						echo __('plugins.importexport.common.errorsOccured') . "\n";
+						$i = 0;
+						foreach ($currentErrors as $currentErrorMessages) {
+							if (count($currentErrorMessages) > 0) {
+								echo ++$i . '.' . __($localeKey) . "\n";
+								foreach ($currentErrorMessages as $currentErrorMessage) {
+									echo '- ' . $currentErrorMessage . "\n";
+								}
+							}
+						}
+						$foundErrors = true;
+					}
+				}
+				// If there are any data or validataion errors
+				// delete imported objects.
+				if ($foundErrors || !empty($validationErrors)) {
+					// remove all imported issues and sumissions
+					foreach (array_keys($errorTypes) as $assocType) {
+						$deployment->removeImportedObjects($assocType);
+					}
+				}
+				return;
+			case 'export':
+				$outputDir = dirname($xmlFile);
+				if (!is_writable($outputDir) || (file_exists($xmlFile) && !is_writable($xmlFile))) {
+					echo __('plugins.importexport.common.cliError') . "\n";
+					echo __('plugins.importexport.common.export.error.outputFileNotWritable', array('param' => $xmlFile)) . "\n\n";
+					$this->usage($scriptName);
+					return;
+				}
+				if ($xmlFile != '') switch (array_shift($args)) {
```

Comment:
This code still needs to be adapted -- `article`/`articles` is the OJS-specific term.